dart-mode.el

```diff
@@ -544,6 +545,15 @@ server sends a response to a request, it tags the response with the ID of the
 request.  We look up the callback for the request in this alist and run it with
 the JSON decoded server response.")
 
+(defvar dart--analysis-completion-callbacks  nil
```

Comment:
Unless the IDs overlap (which I imagine they don't), I'd prefer to have all the callbacks in a single alist.